src/v2/ISuperstateTokenV2.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import {IERC20Upgradeable} from "openzeppelin-contracts-upgradeable/interfaces/IERC20Upgradeable.sol";
+import {IERC7246} from "src/interfaces/IERC7246.sol";
+import {AllowList} from "src/AllowList.sol";
+
+interface ISuperstateTokenV2 is IERC20Upgradeable, IERC7246 {
```

Comment:
We may want to consider preserving `ISuperstateToken`, and then just have additional interfaces for any deltas over versions?

The benefit of having a consistent `ISuperstateToken` is it allows us to easily make unit tests generalizable over token versons. See `SuperstateTokenTestBase` and `SuperstateTokenStorageLayoutTestBase` for examples of how this is done.

Particularly:

```
    ISuperstateToken public oldToken;
    ISuperstateToken public newToken;
    ISuperstateToken public currentToken;
```

Reply:
I don't have a strong opinion. I think having the version numbers for the token contracts but not for the interface is marginal

Reply:
cc @maxwolff any thoughts here?

Reply:
dont have too strong an opinion but agree w chris above. makes it a bit harder to forget to update the right lines of the tests, since "ISuperstateToken" is always latest.